lib/AST/Decl.cpp

```diff
@@ -9464,6 +9464,19 @@ GenericTypeParamDecl *OpaqueTypeDecl::getExplicitGenericParam(
   return genericParamType->getDecl();
 }
 
+bool OpaqueTypeDecl::exportUnderylingTypeToClients() const {
+  auto mod = getDeclContext()->getParentModule();
+  if (mod->getResilienceStrategy() != ResilienceStrategy::Resilient)
+    return true;
+
+  ValueDecl *namingDecl = getNamingDecl();
+  if (namingDecl->getAttrs().hasAttribute<AlwaysEmitIntoClientAttr>() ||
```

Comment:
`@_transparent` and `@backDeployed` also expose the body of the function to clients so I would expect to see checks for those attributes as well. Should we check the resilience expansion instead of looking for specific attributes?

Reply:
Also, I've learned that there are sometimes special SIL functions that bypass all the normal logic and get serialized regardless of their attributes or resilience expansion. So perhaps the more definitive check would be to just directly check if the function is serialized or not, rather than attempting to reverse engineer if it would be.

Reply:
Yeah, I'd prefer that. It's what the logic used to rely on, but for a computed property we just have access to the `VarDecl` which doesn't offer that service. Maybe it should but this would be the only client as it attaches to the var instead of the accessors.

Reply:
I see the difficulty now. Perhaps we should have different logic for functions and var decls so that we continue using the most robust approach we can for functions while also closing the gap more for var decls.

Reply:
Hm, there's also another problem here, which is that the attributes may not be written on the var decl. You can write the attributes explicitly on the accessors. I think a fully correct approach may need to involves something like enumerating all of the accessors for the var decl and checking the resilient expansions for all of them.

Reply:
Good point, I'll look into that.

For my first question, it looks like backDeployed is not supported with opaque return types: `warning: '@backDeployed' is unsupported on a instance method with a 'some' return type`

Reply:
> it looks like backDeployed is not supported with opaque return types
That's a temporary restriction though, I expect to lift it.